### ISSUE TITLE
Creating XCFramework that uses Cocoapods dependencies when used generates "implemented in both" warning. 

### ISSUE THREAD

**Issue by felipeflorencio** [CocoaPods/CocoaPods#11433]
Hi,

I'm creating an SDK using Xcode XCframework for it.

This SDK, has as dependencies Pods, that I need to have in my framework when used.

I generate my xcframework successfully, I have created a Podspec to distribute it, and in it I add the dependencies that this project needs.

When installed, and run, I always get warnings like these for multiple libraries:
`objc[23876]: Class CvVideoCamera is implemented in both /private/var/containers/Bundle/Application/41369DAF-1955-469D-8681-2B398F9D5853/SDKImplementation.app/Frameworks/SDK.framework/SDK (0x10a528eb0) and /private/var/containers/Bundle/Application/41369DAF-1955-469D-8681-2B398F9D5853/SDKImplementation.app/SDKImplementation (0x1039e4750). One of the two will be used. Which one is undefined.
`

Not sure if the problem is how I setup my Project or Cocoapods.

I was able to remove this from those who use it using a script to remove OTHER_LD_FLAG with the frameworks that are saying that is duplicated.

This is my SDK pod file for example:
```
# Uncomment the next line to define a global platform for your project
platform :ios, '12.0'

target 'SDK' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  pod 'OpenCV'
  pod 'AWSCore'
  pod 'AWSS3'
  pod 'NYTPhotoViewer'
  pod 'GoogleMLKit/BarcodeScanning', '2.6.0'
  pod 'OpenCV'
  pod 'TOCropViewController'
  pod 'ScanditBarcodeCapture'
  pod 'ScanditCaptureCore'

end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['BUILD_LIBRARY_FOR_DISTRIBUTION'] = 'YES'
    end
  end
end
```

For me looks correct.

And this is the PodSpec that I'm using to distribute this XCframework:

```
Pod::Spec.new do |spec|
    spec.name               = "SDK"
    spec.version            = "0.0.1"
    spec.summary            = " SDK"
    spec.description        = "SDK for validation Sneakers"
    spec.homepage           = "https://www.mysite.com"
    spec.documentation_url  = "..."
    spec.license            = { :type => "MIT" }
    spec.author             = { "SDK Company" => "..." }
    spec.swift_version      = "5.3"
    spec.platform           = :ios, "12.0"
    spec.source             = { :path => '.' }

    # Published binaries
    spec.vendored_frameworks = "SDK.xcframework"
  
    # Dependency
    spec.dependency 'AWSCore'
    spec.dependency 'AWSS3'
    spec.dependency 'NYTPhotoViewer'
    spec.dependency 'GoogleMLKit/BarcodeScanning', '2.6.0'
    spec.dependency 'OpenCV'
    spec.dependency 'TOCropViewController'
    spec.dependency 'ScanditBarcodeCapture'
    spec.dependency 'ScanditCaptureCore'
  end
  ```

**Comment by AdamShi** [comment]
Hi, how did you solve this problem?

**Comment by felipeflorencio** [comment]
No other way as far as I know to fix this rather than "vendoring" the dependencies.

I did not use the dependencies as cocoapods pods and embbed in the framework itself.

This doesn't fix the problem, I had to get the source code and "prepend" all the class names and files for those.

For those that I did not have access to the source code because as a close source, I had to get the static version of it, the downside for it is that static frameworks are twice the size so they can be used on simulators and devices.

This is the only solution, and Is the solution "found" for example by AWS a long time ago.
